tests/test_parser.py

```diff
@@ -257,6 +259,96 @@ def test_convert_to_date_cast_result_to_date(self, mocked_locale,
                          expected_output)
 
 
+class Test_bs_safeget(unittest.TestCase):
+    def test_bs_safeget_return_expected_text(self):
+        mocked_soup = MagicMock()
+        selector = 'some_tag'
+        mocked_soup.select.return_value = ['some text']
+        self.assertEqual(bs_safeget(mocked_soup, selector), 'some text')
+
+    def test_bs_safeget_return_false_when_selector_is_missing(
+            self):
+        mocked_soup = Mock()
+        mocked_soup.select.return_value = ''
+        selector = 'h1'
+        self.assertFalse(bs_safeget(mocked_soup, selector))
+
+
+@patch('smarsy.parse.bs_safeget')
+class TestCreateParentsDict(unittest.TestCase):
+    def test_bs_safeget_called_with_expected_html_for_img_getting(
+            self, mocked_get):
+        tag1, tag2 = '[valign=top]', 'img[src]'
+        create_parents_dict('html')
+        mocked_get.assert_called_with('html', tag1, tag2)
+
+    def test_create_parents_dict_return_false_with_none_img(self, mocked_get):
+        mocked_get.return_value = None
+        self.assertFalse(create_parents_dict('html'))
+
+    @patch('builtins.hasattr')
+    def test_bs_safeget_return_expected_img(self, mock_print, mocked_get):
+        mocked_get.return_value = 'some_img.jpg'
+        create_parents_dict('html')
+        self.assertTrue(mock_print.call_count, 1)
+
+
+class TestParseParentPage(unittest.TestCase):
+    @patch('smarsy.parse.BeautifulSoup')
+    def test_parent_page_content_called_with_expected_html(self, mocked_soup):
+        html = '<tr></tr>'
+        parent_page_content_to_object(html)
+        mocked_soup.assert_called_with(html, 'html.parser')
+
+    @patch('bs4.BeautifulSoup')
```

Comment:
this mock is not working. I mean that it is mocking BS in test module, not in parse. If you change decorator to patch any other class/method it will still work, try ` @patch('smarsy.parse.bs_safeget')`